BridgeAppSDK/SBAGenericStepDataSource.swift

```diff
@@ -0,0 +1,614 @@
+ manages and stores answers provided thru user input, and provides an ORKResult with those anwers upon request.
+ 
+ It also provides several convenience methods for saving or selecting answers, checking if all answers are valid,
+ and retrieving specific model objects that may be needed by the ViewController.
+ 
+ The tableView data source is comprised of 3 objects:
+ 
+ 1) SBAGenericStepTableSection - An object representing a section in the tableView. It has one or more SBAGenericStepTableItemGroups.
+ 
+ 2) SBAGenericStepTableItemGroup - An object representing a specific question supplied by ORKStep in the form of ORKFormItem.
+ An ORKFormItem can have multiple answer options, such as a boolean question or text choice
+ question. Or, it can have just one answer option, in the case of alpha/numeric questions.
+ 
+ Upon init(), the ItemGroup will create one or more SBAGenericStepTableItem representing the answer
+ options for the ORKFormItem. The ItemGroup is responsible for storing/computing the answers
+ for its ORKFormItem.
+ 
+ 3) SBAGenericStepTableItem - An object representing a specific answer option from the ItemGroup (ORKFormItem), such as a Yes or No
+ choice in a boolean question or a string or number that's entered thru a text field. There will be
+ one TableItem for each indexPath in the tableView.
+ */
+
+public protocol SBAGenericStepDataSourceDelegate {
+    func answersDidChange()
+}
+
+open class SBAGenericStepDataSource: NSObject {
+    
+    open var delegate: SBAGenericStepDataSourceDelegate?
+    open var sections: Array<SBAGenericStepTableSection> = Array()
+    open var step: ORKStep?
+    
+    /**
+     Initialize a new SBAGenericStepDataSource.
+     @param  step       The ORKStep
+     @param  result     The previous ORKResult, if any
+     */
+    public init(step: ORKStep?, result: ORKResult?) {
+        super.init()
+        self.step = step
+        populate()
+        if result != nil {
+            updateAnswers(from: result!)
+        }
+    }
+    
+    func updateAnswers(from result: ORKResult) {
+        
+        // iterate the provided results, find the corresponding ItemGroup for each, and save the answer
+        // provided in the results
+        
+        if let stepResult = result as? ORKStepResult {
+            if let results = stepResult.results as? Array<ORKQuestionResult> {
+                for result in results {
+                    let answer = result.answer ?? ORKNullAnswerValue()
+                    if let group = itemGroup(with: result.identifier) {
+                        group.answer = answer as AnyObject
+                    }
+                }
+            }
+        }
+    }
+    
+    
+    public func updateDefaults(_ defaults: NSMutableDictionary) {
+        
+        // TODO: Josh Bruhin, 6/12/17 - implement. this may require access to a HealthKit source.
+        // This has not yet been needed for the onboarding flow
```

Comment:
Sorry, reading default HealthKit values to update the model. In RK, this is all done in private methods -> (void)refreshDefaults in ORKFormStepVC. Anywhere in Bridge I can access this data? I've been looking, but no luck.